rebar.config.erl

```diff
@@ -457,6 +470,22 @@ list_dir(Dir) ->
     {ok, Names} = file:list_dir(Dir),
     [list_to_atom(Name) || Name <- Names, filelib:is_dir(filename:join([Dir, Name]))].
 
+get_compile_secret(SecretFile) ->
+    {ok, Secret} = file:read_file(SecretFile),
+    string:strip(binary_to_list(Secret), right).
+
+crypto_key(init) -> ok;
+crypto_key(key) -> ok;
+crypto_key({debug_info, _Mode, _Module, _Filename}) ->
+    case get('__emqx_compile_secret_file__') of
+        undefined ->
+            Res = get_compile_secret(os:getenv("EMQX_COMPILE_SECRET_FILE")),
```

Comment:
do we have to read the file inside the closure ?

Reply:
this worked for me:
```
make_fun(Secret) ->
    fun(init) -> ok;
       (key) -> ok;
       ({debug_info, _Mode, _Module, _Filename}) -> Secret
    end.
```

Reply:
what do you prefer?

Reply:
I prefer side-effect free crypto_key callback.